app/controllers/admin/borrow_books_controller.rb

```diff
@@ -40,58 +43,98 @@ def show
-    if @borrow_book.save
-      flash[:success] = t "message.borrow_books.new"
-      redirect_to admin_borrow_book_path(@borrow_book), status: :see_other
+  def cancel
+    if @borrow_book.update status: :cancel,
+                           reason: params[:borrow_book][:reason]
+      send_status_change_email @borrow_book, "cancel"
+      flash[:success] = t "message.borrow_books.canceled"
     else
-      render :new, status: :unprocessable_entity
+      flash[:danger] = t "message.borrow_books.status_fail"
     end
+    redirect_to borrow_admin_borrow_books_path
   end
 
-  def edit
-    @breadcrumb_items = [
-      {name: t(".borrow.title"), url: admin_borrow_books_path},
-      {name: @borrow_book.id, url: admin_borrow_book_path(@borrow_book)},
-      {name: t(".edit.title")}
-    ]
+  def returned
+    episode = @borrow_book.episode
+    if episode.increment!(:qty) && @borrow_book.update(status: :returned)
+      activate_overdue_user
+      send_status_change_email @borrow_book, "returned"
+      flash[:success] = t "message.borrow_books.returned"
+    else
+      flash[:error] = t "message.borrow_books.status_fail"
+    end
+    redirect_to return_admin_borrow_books_path
   end
 
-  def update
-    if @borrow_book.update(borrow_book_params)
-      flash[:success] = t "message.borrow_books.updated"
-      redirect_to admin_borrow_book_path(@borrow_book)
+  def lost
+    user = @borrow_book.borrow_card.user
+
+    if @borrow_book.lost!
+      user.update lost_time: user.lost_time + 1, activated: false
+      send_status_change_email @borrow_book, "lost"
+      flash[:success] = t "message.borrow_books.lost"
     else
-      render :edit, status: :unprocessable_entity
+      flash[:error] = t "message.borrow_books.status_fail"
     end
+
+    redirect_to return_admin_borrow_books_path
   end
 
-  def destroy
-    if @borrow_book.destroy
-      flash[:success] = t "message.borrow_books.deleted"
-    else
-      flash[:danger] = t "message.borrow_books.delete_fail"
+  def refresh
+    overdue_borrow_books = BorrowBook.confirm.select do |borrow_book|
+      borrow_book.due_date < Time.current
+    end
+
+    overdue_borrow_books.each do |borrow_book|
+      if borrow_book.overdue!
+        borrow_book.borrow_card.user.update activated: false
+        send_status_change_email borrow_book, "overdue"
+      else
+        flash[:error] = t "message.borrow_books.status_fail"
+      end
     end
-    redirect_to borrow_admin_borrow_books
+
+    flash[:success] = t "message.borrow_books.refresh_success"
+    redirect_to return_admin_borrow_books_path
   end
 
   private
 
   def load_borrow_book
-    @borrow_book = BorrowBook.find(params[:id])
+    @borrow_book = BorrowBook.find params[:id]
     return if @borrow_book
 
     flash[:danger] = t "message.borrow_books.not_found"
     redirect_to admin_borrow_books_path
   end
+
+  def activate_overdue_user
+    return unless @borrow_book.status_before_last_save == "overdue"
```

Comment:
```suggestion
    return unless @borrow_book.status_before_last_save.to_sym ==  :overdue
```
Những phần này nên dùng symbol ạ.